15-components-lab_v16/src/01-timeconverter.jsx

```diff
@@ -8,5 +8,38 @@
 
 import React from 'react';
 
-export default React.createClass({
-});
+
+export default class TimeConverter extends React.Component {
+
+    constructor(props){
+        super(props);
+        this.state = { hours : 1, minutes: 60, seconds: 3600}
+    }
+
+    hourUpdate = (e) => {
+        const val = this.minutes.value;
+        this.setState(oldState => ({ hours: val/60, minutes : val, seconds: val * 60 }));
+    }
+
+    minUpdate = (e) => {
+        const val = this.hours.value;
+        this.setState(oldState => ({ hours: val, minutes : val *60, seconds: val * 3600 }));
+    }
+
+    secUpdate = (e) => {
+        const val = this.seconds.value;
+        this.setState(oldState => ({ hours: val/3600, minutes : val/60, seconds: val }));
+    }
+
+
+    render(){
+        var boxStyle = {marginRight : '5px', width: '40px'}
+        return (
```

Comment:
This just looks bad. It's really hard to read this render code let alone maintain it later. Here's the same code with line breaks:

```markup
          <div>
            <span>H</span>
            <input
              style={boxStyle}
              type="text"
              value={this.state.hours}
              ref={(hours) => this.hours = hours}
              onChange={this.minUpdate}
            />
            <span>M</span>
            <input
              style={boxStyle}
              type="text"
              value={this.state.minutes}
              ref={(minutes) => this.minutes = minutes}
              onChange={this.hourUpdate}
            />  
            <span>S</span>
            <input
              style={boxStyle}
              type="text"
              value={this.state.seconds}
              ref={(seconds) => this.seconds = seconds}
              onChange={this.secUpdate}/>
          </div>
```

But we can do even better. For example we can write another function that would return the span and input:

```jsx
    renderField(field) {
      const labelText = field[0].toUpperCase();
      const ref = (val) => this[field] = val;
      const updater = this[field + 'Update'];
      const boxStyle = {marginRight : '5px', width: '40px'};
      
      return (
        <label>
            <span>{labelText}</span>
            <input
              style={boxStyle}
              type="text"
              value={this.state[field]}
              ref={ref}
              onChange={updater}
            />          
        </label>
      )
    }
    
    render(){
        
        return (       
          <div>
            {this.renderField('hours')}
            {this.renderField('minutes')}
            {this.renderField('seconds')}
          </div>
        )
    }
```

Reply:
I noticed some naming mismatch in your field value - for example when changing the minutes you'll call hourUpdate. This was a bit confusing to read.